File: Python/0243-e-shortest.word-distance.py
class Solution:
    def shortestDistance(self, words: List[str], word1: str, word2: str) -> int:
        pos1 = -1
        pos2 = -1
        dist = len(words)
        for i in range(len(words)):
            if words[i] == word1:
                pos1 = i
            elif words[i] == word2:
                pos2 = i
        
            if pos1 != -1 and pos2 != -1:
                dist = min(dist ,abs(pos1 - pos2))
        return dist

File: leetcode/C++/0243-e-shortest-word-distance.cpp
class Solution {
public:
    int shortestDistance(vector<string>& words, string word1, string word2) {
        int min_d = words.size() - 1;
        int p1 = -1, p2 = -1;
        for (int i = 0; i < words.size(); i++) {
            if (words[i] == word1)
                p1 = i;
            if (words[i] == word2)
                p2 = i;
            if (p1 != -1 && p2 != -1)
                min_d = min(min_d, abs(p1 - p2));
        }
        return min_d;
    }
};